### PR TITLE
CASMMON-247 adding metrics for web hook stages 

### PR DESCRIPTION
**Pull request by rambabubolla**
Added stub operations to get prom-metrics for post-install-service-check and post-install-check stages
currently, iuf operation does not provide an inbuilt mechanism for r Prometheus metrics.
At the first step added epoch start and end time for extract-release-distributions operation. we are planing to use these metrics for creating grafana iuf timing dashboard.

Please note we are unable to create WorkflowTemplate for the Prometheus metrics. So we have used it directly in extract-release-distributions.yaml.

Testing: tested on frigg.

curl 10.43.0.244:9090/metrics |grep post-install-check
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0argo_workflows_operation_counter{opname="post-install-check-end",pname="cpe",pversion="22.12.8-0dfee16",stage="post-install-check",status="Succeeded",type="product"} 2
argo_workflows_operation_counter{opname="post-install-check-start",pname="cpe",pversion="22.12.8-0dfee16",stage="post-install-check",status="Succeeded",type="product"} 6
argo_workflows_operation_time{opend="1674556210",opname="post-install-check-start",opstart="1674556200",pdname="cpe",pdversion="22.12.8-0dfee16",stage="post-install-check",type="product"} 10
argo_workflows_operation_time{opend="1674556676",opname="post-install-check-start",opstart="1674556664",pdname="cpe",pdversion="22.12.8-0dfee16",stage="post-install-check",type="product"} 12
argo_workflows_operation_time{opend="1674562012",opname="post-install-check-start",opstart="1674562002",pdname="cpe",pdversion="22.12.8-0dfee16",stage="post-install-check",type="product"} 10
argo_workflows_operation_time{opend="1674563032",opname="post-install-check-start",opstart="1674563022",pdname="cpe",pdversion="22.12.8-0dfee16",stage="post-install-check",type="product"} 10
argo_workflows_operation_time{opend="1674563288",opname="post-install-check-start",opstart="1674563278",pdname="cpe",pdversion="22.12.8-0dfee16",stage="post-install-check",type="product"} 10
argo_workflows_operation_time{opend="1674563320",opname="post-install-check-end",opstart="1674563310",pdname="cpe",pdversion="22.12.8-0dfee16",stage="post-install-check",type="product"} 10
argo_workflows_operation_time{opend="1674630524",opname="post-install-check-start",opstart="1674630514",pdname="cpe",pdversion="22.12.8-0dfee16",stage="post-install-check",type="product"} 10
argo_workflows_operation_time{opend="1674630556",opname="post-install-check-end",opstart="1674630546",pdname="cpe",pdversion="22.12.8-0dfee16",stage="post-install-check",type="product"} 10
100  351k    0  351k    0     0  17.1M      0 --:--:-- --:--:-- --:--:-- 17.1M
ncn-m001:/opt/cray/iuf/rambabu # curl 10.43.0.244:9090/metrics |grep post-install-service-check
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  351k    0  351k    0     0  21.4M      0 --:--:-- --:--:-- --:--:-- 21.4M
ncn-m001:/opt/cray/iuf/rambabu # curl 10.43.0.244:9090/metrics |grep post-install-service-check
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0argo_workflows_operation_counter{opname="post-install-service-check-end",pname="cpe",pversion="22.12.8-0dfee16",stage="post-install-service-check",status="Succeeded",type="product"} 1
argo_workflows_operation_counter{opname="post-install-service-check-start",pname="cpe",pversion="22.12.8-0dfee16",stage="post-install-service-check",status="Succeeded",type="product"} 1
argo_workflows_operation_time{opend="1674631235",opname="post-install-service-check-start",opstart="1674631224",pdname="cpe",pdversion="22.12.8-0dfee16",stage="post-install-service-check",type="product"} 11
argo_workflows_operation_time{opend="1674631266",opname="post-install-service-check-end",opstart="1674631256",pdname="cpe",pdversion="22.12.8-0dfee16",stage="post-install-service-check",type="product"} 10
100  352k    0  352k    0     0  21.4M      0 --:--:-- --:--:-- --:--:-- 21.4M

### CODE REVIEW

**Review by rustydb — APPROVED** [review]

File: workflows/iuf/operations/post-install-check-end.yaml [hunk 1]
```diff
@@ -0,0 +1,114 @@
+#
+# MIT License
+#
+# (C) Copyright 2022-2023 Hewlett Packard Enterprise Development LP
```
Comment (mharding-hpe) [inline workflows/iuf/operations/post-install-check-end.yaml:4]: If this is a new file, should 2022 be included in the copyright header?

File: workflows/iuf/operations/post-install-check-start.yaml [hunk 2]
```diff
@@ -0,0 +1,114 @@
+#
+# MIT License
+#
+# (C) Copyright 2022-2023 Hewlett Packard Enterprise Development LP
```
Comment (mharding-hpe) [inline workflows/iuf/operations/post-install-check-start.yaml:4]: If this is a new file, should 2022 be included in the copyright header?

File: workflows/iuf/operations/post-install-service-check-end.yaml [hunk 3]
```diff
@@ -0,0 +1,114 @@
+#
+# MIT License
+#
+# (C) Copyright 2022-2023 Hewlett Packard Enterprise Development LP
```
Comment (mharding-hpe) [inline workflows/iuf/operations/post-install-service-check-end.yaml:4]: If this is a new file, should 2022 be included in the copyright header?

File: workflows/iuf/operations/post-install-service-check-start.yaml [hunk 4]
```diff
@@ -0,0 +1,114 @@
+#
+# MIT License
+#
+# (C) Copyright 2022-2023 Hewlett Packard Enterprise Development LP
```
Comment (mharding-hpe) [inline workflows/iuf/operations/post-install-service-check-start.yaml:4]: If this is a new file, should 2022 be included in the copyright header?

**Review by mharding-hpe — APPROVED** [review]

File: workflows/iuf/operations/post-install-service-check-start.yaml [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+              value: "{{=jsonpath(inputs.parameters.global_params, '$.product_manifest.current_product.manifest.version')}}"
+            - key: status
+              value: "{{status}}"
+          counter:
+            value: "1"
+    inputs:
+      parameters:
+      - name: global_params
+    steps:
+    - - name: start-operation
```
Comment (rustydb) [inline workflows/iuf/operations/post-install-service-check-start.yaml:57]: Do these really need double-dashes?

**Review by rustydb — CHANGES_REQUESTED** [review]

File: workflows/iuf/operations/post-install-service-check-start.yaml (hunk 5 above)
Comment (mharding-hpe) [reply workflows/iuf/operations/post-install-service-check-start.yaml:57]: Good catch -- that does look odd.

Comment (rambabubolla) [reply workflows/iuf/operations/post-install-service-check-start.yaml:57]: @mharding-hpe  yes we need -- for steps exaction in argo workflow syntax

**Review by rustydb — APPROVED** [review]

**Pull request merged**